lib/modules/manager/gradle/extract/catalog.ts

```diff
@@ -184,19 +184,23 @@ function extractDependency({
       skipReason,
     };
   }
-
+  const versionRef = isVersionPointer(descriptor.version)
+    ? descriptor.version.ref
+    : null;
   if (isArtifactDescriptor(descriptor)) {
-    const { group: groupName, name } = descriptor;
+    const { group, name } = descriptor;
+    const groupName = is.nullOrUndefined(versionRef) ? group : versionRef; // usage of common variable should have higher priority than other values
```

Comment:
Does it mean we always have some grouping, i.e. `com.example` for `com.example:foo:1.2.3` and `bar` for `com.example:foo:${bar}`? If so, do we want to extend this behavior to `build.gradle` and `gradle.properties` parsing as well?

Reply:
@zharinov Yes groupName will be the variable name in case where there is a variable. (yes correct to your example)
As for the second question I think yes. wdyt @rarkins ?

Reply:
@zharinov 

`build.gradle`:
- Checked the code and it seems you were already setting the `groupName` values to variables when there are variable substitutions in the dependencies block of `build.gradle`. Here is a test for it https://github.com/hasanwhitesource/renovate/blob/5946790b6601471363ab06907638d3b9ec81a862/lib/modules/manager/gradle/parser.spec.ts#L62
- After handling plugins in this pr we will also handle plugins in `build.gradle`

As for the `gradle.properties` could there be a reference to a variable that is defined in the same file ? Kind of new to the gradle files but looked around a bit and saw that users just define their properties to be used there.

Reply:
Initially, I've assumed `gradle.properties` to contain only variable definitions.
But in some cases people define the entire coordinates, e.g. `com.example:foo:1.2.3`.
I don't think variable references are possible inside `gradle.properties`, so it should be okay as is.

Reply:
In order to keep the scope of this PR smaller, it seems reasonable to delay decision about `groupName` fields for `build.gradle` files, implementing it in separate PR if needed.

Reply:
I will only keep the logic for toml parsing then 👍🏻